JASP-Engine/JASP/R/MixedModelsCommon.R

```diff
@@ -97,50 +97,78 @@
 }
 
 ### common mixed-models functions
-.mmReadData      <- function(dataset, options, type = "LMM") {
-  if (!is.null(dataset)) {
-    return(dataset)
-  } else{
+.mmReadData      <- function(jaspResults, dataset, options, type = "LMM") {
+  if (is.null(dataset)) {
     if (type %in% c("LMM","BLMM")) {
-      return(readDataSetToEnd(
+      dataset <- readDataSetToEnd(
         columns.as.numeric = options$dependentVariable,
         columns = c(
           options$fixedVariables,
           options$randomVariables
         )
-      ))
+      )
     } else if (type %in% c("GLMM","BGLMM")) {
       if (options$family == "binomial_agg"){
-        return(readDataSetToEnd(
+        dataset <- readDataSetToEnd(
           columns.as.numeric = c(options$dependentVariable, options$dependentVariableAggregation),
           columns = c(
             options$fixedVariables,
             options$randomVariables
           )
-        ))
+        )
       } else  if (options$dependentVariableAggregation == "") {
-        return(readDataSetToEnd(
+        dataset <- readDataSetToEnd(
           columns.as.numeric = options$dependentVariable,
           columns = c(
             options$fixedVariables,
             options$randomVariables
           )
-        ))
+        )
       }  
     }
   }
+
+  dataset <- data.frame(dataset)
+
+  # check and use only the variables that actually used for modeling
+  used_variables <- .v(c(
```

Comment:
This is some what hard to read. And no space after if, and no space after a comma.